Source/SwiftLintFramework/Rules/ShorthandOperatorRule.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  ShorthandOperatorRule.swift
+//  SwiftLint
+//
+//  Created by Marcelo Fabri on 01/06/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct ShorthandOperatorRule: ConfigurationProviderRule {
+
+    public var configuration = SeverityConfiguration(.error)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "shorthand_operator",
+        name: "Shorthand Operator",
+        description: "Prefer shorhand operators (+=, -=, *=, /=) over doing the operation and assigning.",
+        nonTriggeringExamples: allOperators.flatMap { operation in
+            [
+                "foo \(operation)= 1",
+                "foo \(operation)= variable",
+                "foo \(operation)= bar.method()",
+                "self.foo = foo \(operation) 1",
+                "foo = self.foo \(operation) 1"
+            ]
+        },
+        triggeringExamples: allOperators.flatMap { operation in
+            [
+                "↓foo = foo \(operation) 1\n",
+                "↓foo = foo \(operation) aVariable\n",
+                "↓foo = foo \(operation) bar.method()\n",
+                "↓foo = 1 \(operation) foo\n",
+                "↓foo = aVariable \(operation) foo\n",
+                "↓foo = bar.method() \(operation) foo\n",
+                "↓foo = bar.method(param: 1, otherParam: 2) \(operation) foo\n",
+                "↓foo.aProperty = foo.aProperty \(operation) 1\n",
+                "↓self.aProperty = self.aProperty \(operation) 1\n"
+            ]
+        }
+    )
+
+    private static let allOperators = ["+", "-", "/", "*"]
+
+    private static let pattern: String = {
+        let escapedOperators = allOperators.map { "\\\($0)" }.joined()
+        let operators = "[\(escapedOperators)]"
+        let operand = "[\\w\\d\\.]+?"
+        let spaces = "[^\\S\\r\\n]*?"
+        let otherOperand = "\(spaces).+?\(spaces)"
+
+        let pattern1 = "^\(spaces)(\(operand))\(spaces)=\(spaces)(\\1)\(spaces)\(operators)"
+        let pattern2 = "^\(spaces)(\(operand))\(spaces)=\(otherOperand)\(operators)\(spaces)(\\3)"
+
+        return "\(pattern1)|\(pattern2)"
+    }()
+
+    // swiftlint:disable:next force_try
+    private static let regex = try! NSRegularExpression(pattern: pattern, options: [.anchorsMatchLines])
```

Comment:
not _that_ complicated. We store a bitmask of the options as the key in a multimap.